### PR TITLE
[DevOps] Feature/ basic cmake configuration and source tree

### PR DESCRIPTION
**Pull request by jeremyelalouf**

### CODE REVIEW

**Review by Snowy1803 — COMMENTED** [review]
- Retour à la ligne en fin de fichiers
- Mets les dependences ( ASTGen utilise lib AST, logiquement)

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required (VERSION 3.29)
+
+set(GLU_CMAKE_DIR "${CMAKE_CURRENT_LIST_DIR}/cmake" CACHE PATH "The path to the cmake directory")
+list(APPEND CMAKE_MODULE_PATH ${GLU_CMAKE_DIR})
+
+# CMake will prevent in-source builds
+set(CMAKE_DISABLE_IN_SOURCE_BUILD YES)
+
+project(Glu C CXX)
```
Comment (Snowy1803) [inline CMakeLists.txt:9]: C?

File: cmake/UnixCompileRules.cmake [hunk 2]
```diff
@@ -0,0 +1,3 @@
+set (CMAKE_CXX_FLAGS_COMMON "-Wall -Wextra")
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON} -O1 -g")
```
Comment (Snowy1803) [inline cmake/UnixCompileRules.cmake:2]: The `-Og` level specifies we want to run optimizations that don't degrade debuggability (similar to -O1)
```suggestion
set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON} -Og -g")
```

File: tools/glud/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+add_executable (glud)
```
Comment (Snowy1803) [inline tools/glud/CMakeLists.txt:1]: you can probably remove glud

File: lib/IRGen/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+add_library (IRGen SHARED)
+
+target_include_directories(IRGen
+    PUBLIC
+    ${PROJECT_SOURCE_DIR}/include/IRGen
+)
+
+target_link_libraries(IRGen
+    PUBLIC
+    GIL
```
Comment (Snowy1803) [inline lib/IRGen/CMakeLists.txt:10]: we'll need LLVM

File: tools/glud/CMakeLists.txt (hunk 3 above)
Comment (jeremyelalouf) [reply tools/glud/CMakeLists.txt:1]: Indeed !

**Review by Snowy1803 — APPROVED** [review]

**Pull request merged**